air/src/aux_table/mod.rs

```diff
@@ -36,6 +39,8 @@ pub fn get_transition_constraint_degrees() -> Vec<TransitionConstraintDegree> {
         .map(|&degree| TransitionConstraintDegree::new(degree))
         .collect();
 
+    degrees.append(&mut bitwise_pow2::get_transition_constraint_degrees());
+
     degrees.append(&mut memory::get_transition_constraint_degrees());
```

Comment:
Yep, fully agreed here. I think it makes sense to revisit things like this during a future optimization pass, since it will make the code less readable and harder to follow. I think for this initial version it makes sense to have things that are very explicit, easy to test, and easy to convert into a tool that reads high-level constraint descriptions and generates optimized code from them (as discussed).